File: 202-happy-number/happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int num=n;
        num= Add(num);
        while (num>1 && num!=4)
        {
            num= Add(num);
        }
        return (num==1)? true:false;
    }
    public int Add(int num){
        int sum=0, a;
        while (num>0)
        {
            a= num%10;
            sum= sum+ (a*a);
            num= num/10;
        }
        return sum;
    }
}

File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        n= self.Add(n)
        while (n>1 and n!=4):
            n=self.Add(n)
        return n==1

    def Add(self, num):
        sum=0
        while num>0:
            a= num%10
            sum= sum+ (a*a)
            num= num//10
        return sum